app/controllers/programs_controller.rb

```diff
@@ -67,8 +72,15 @@ def create
 
   # PATCH/PUT /programs/1 or /programs/1.json
   def update
+    uniqname = program_params[:instructor_attributes][:uniqname]
+    if ProgramManager.find_by(uniqname: uniqname).present?
```

Comment:
Why couple the creation on ProgramManager here in the Programs controller? Pass this to the ProgramManager controller.

Reply:
This is a program instructor. The program form has a nested form to create an instructor
```
<div class="my-5">
    <%= form.fields_for :instructor, @instructor do |instructor_attributes| %>
      <%= form.label :uniqname, "Instructor's uniqname *", class: "fancy_label" %>
      <%= instructor_attributes.text_field :uniqname, required: true, placeholder: "Uniqname", class: "input_text_field" %>
    <% end %>
</div>
```
The instructor is located in the program_managers table:
```
class Program < ApplicationRecord
  belongs_to :instructor, class_name: 'ProgramManager', foreign_key: :instructor_id
```